crates/pera-core/src/dwallet_mpc/mpc_party.rs

```diff
@@ -0,0 +1,111 @@
+use crate::dwallet_mpc;
+use crate::dwallet_mpc::dkg::{DKGFirstParty, DKGSecondParty};
+use crate::dwallet_mpc::network_dkg::advance_network_dkg;
+use crate::dwallet_mpc::presign::{PresignFirstParty, PresignSecondParty};
+use crate::dwallet_mpc::sign::SignFirstParty;
+use commitment::CommitmentSizedNumber;
+use dwallet_mpc_types::dwallet_mpc::{DWalletMPCNetworkKey, MPCMessage, MPCPublicInput};
+use group::PartyID;
+use mpc::WeightedThresholdAccessStructure;
+use pera_types::base_types::ObjectID;
+use pera_types::dwallet_mpc_error::DwalletMPCResult;
+use std::collections::HashMap;
+
+pub(super) type AsyncProtocol = twopc_mpc::secp256k1::class_groups::AsyncProtocol;
+
+/// Enum representing the different parties used in the MPC manager.
```

Comment:
```suggestion
/// Enum representing the different parties used in the MPC manager.
/// Note: this structure does not hold actual party instances; instead, it just abstract away type-specific logic by enumerating the different support party types in a single structure.
/// This is used by `MPCManager` and the likes, which we don't want to turn generic as it will interfere with Sui's design.
```